src/test/java/net/snowflake/client/jdbc/BindingDataLatestIT.java

```diff
@@ -122,57 +136,189 @@ public void testTimestampBindingWithNTZType() throws SQLException {
-      statement.execute(
-          "create or replace table regularinsert(ind int, ltz0 timestamp_ltz, tz0 timestamp_tz, ntz0 timestamp_ntz)");
-      statement.execute("alter session set CLIENT_TIMESTAMP_TYPE_MAPPING=TIMESTAMP_LTZ");
-      statement.execute("alter session set TIMEZONE='Asia/Tokyo'");
-      TimeZone.setDefault(TimeZone.getTimeZone("Asia/Tokyo"));
-      Timestamp currT = new Timestamp(System.currentTimeMillis());
-
-      // insert using stage binding
-      PreparedStatement prepStatement =
-          connection.prepareStatement("insert into stageinsert values (?,?,?,?)");
-      statement.execute("ALTER SESSION SET CLIENT_STAGE_ARRAY_BINDING_THRESHOLD = 1");
-      prepStatement.setInt(1, 1);
-      prepStatement.setTimestamp(2, currT);
-      prepStatement.setTimestamp(3, currT);
-      prepStatement.setTimestamp(4, currT);
-      prepStatement.addBatch();
-      prepStatement.executeBatch();
-      statement.execute("ALTER SESSION SET CLIENT_STAGE_ARRAY_BINDING_THRESHOLD = 0");
-
-      // insert using regular binging
-      prepStatement = connection.prepareStatement("insert into regularinsert values (?,?,?,?)");
-      for (int i = 1; i <= 6; i++) {
-        prepStatement.setInt(1, 1);
-        prepStatement.setTimestamp(2, currT);
-        prepStatement.setTimestamp(3, currT);
-        prepStatement.setTimestamp(4, currT);
-        prepStatement.addBatch();
+    TimeZone origTz = TimeZone.getDefault();
+    try (Connection connection = getConnection();
+        Statement statement = connection.createStatement()) {
+      try {
+        statement.execute(
+            "create or replace table stageinsert(ind int, ltz0 timestamp_ltz, tz0 timestamp_tz, ntz0 timestamp_ntz)");
+        statement.execute(
+            "create or replace table regularinsert(ind int, ltz0 timestamp_ltz, tz0 timestamp_tz, ntz0 timestamp_ntz)");
+        statement.execute("alter session set CLIENT_TIMESTAMP_TYPE_MAPPING=TIMESTAMP_LTZ");
+        statement.execute("alter session set TIMEZONE='Asia/Tokyo'");
+        TimeZone.setDefault(TimeZone.getTimeZone("Asia/Tokyo"));
+        Timestamp currT = new Timestamp(System.currentTimeMillis());
+
+        // insert using regular binging
+        try (PreparedStatement prepStatement =
+            connection.prepareStatement("insert into regularinsert values (?,?,?,?)")) {
+          for (int i = 1; i <= 6; i++) {
+            prepStatement.setInt(1, 1);
+            prepStatement.setTimestamp(2, currT);
+            prepStatement.setTimestamp(3, currT);
+            prepStatement.setTimestamp(4, currT);
+            prepStatement.addBatch();
+          }
+          prepStatement.executeBatch();
+        }
+
+        // insert using stage binding
+        try (PreparedStatement prepStatement =
+            connection.prepareStatement("insert into stageinsert values (?,?,?,?)")) {
+          statement.execute("ALTER SESSION SET CLIENT_STAGE_ARRAY_BINDING_THRESHOLD = 1");
```

Comment:
let's move the alter session before the try and have a method for executing prepared statment - it can be the same method as for previous test

Reply:
Done.

Reply:
And, same as above.